src/core/iomgr/fd_posix.c

```diff
@@ -258,25 +260,31 @@ static void set_ready(grpc_fd *fd, gpr_atm *st,
   /* only one set_ready can be active at once (but there may be a racing
      notify_on) */
   int success;
-  grpc_iomgr_closure cb;
+  grpc_iomgr_closure closure;
```

Comment:
What if this was:
```
grpc_iomgr_closure *closure;
```

and set_ready_locked took a `grpc_iomgr_closure**` - so it could simply set the pointer that we already have, and we'd not need to do an allocation here either?